solidity/test/ERC20NoData.ts

```diff
@@ -0,0 +1,201 @@
+    ).to.be.reverted;
+
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(0);
+    expect(await deployedERC20NoData.balanceOf(signerB.address)).to.equal(20);
+    expect(await deployedERC20NoData.balanceOf(signerC.address)).to.equal(20);
+  });
+
+  it('Burn - Signer should burn their own tokens successfully', async function () {
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(0);
+    // Mint tokens to Signer A
+    await expect(
+      deployedERC20NoData.connect(deployerSignerA).mintNoData(deployerSignerA.address, 20),
+    )
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(ZERO_ADDRESS, deployerSignerA.address, 20);
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(20);
+
+    // Signer A burns 5 of their tokens
+    await expect(
+      deployedERC20NoData.connect(deployerSignerA).burnNoData(deployerSignerA.address, 5),
+    )
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(deployerSignerA.address, ZERO_ADDRESS, 5);
+
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(15);
+    // Signer A burns 15 of their tokens
+    await expect(
+      deployedERC20NoData.connect(deployerSignerA).burnNoData(deployerSignerA.address, 15),
+    )
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(deployerSignerA.address, ZERO_ADDRESS, 15);
+
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(0);
+  });
+
+  it("Burn - Signer should not burn another signer's tokens", async function () {
+    expect(await deployedERC20NoData.balanceOf(deployerSignerA.address)).to.equal(0);
+    expect(await deployedERC20NoData.balanceOf(signerB.address)).to.equal(0);
+    expect(await deployedERC20NoData.balanceOf(signerC.address)).to.equal(0);
+    // Signer A mints 20 tokens to itself
+    await expect(
+      deployedERC20NoData.connect(deployerSignerA).mintNoData(deployerSignerA.address, 20),
+    )
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(ZERO_ADDRESS, deployerSignerA.address, 20);
+    // Signer A mints 20 tokens to Signer B
+    await expect(deployedERC20NoData.connect(deployerSignerA).mintNoData(signerB.address, 20))
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(ZERO_ADDRESS, signerB.address, 20);
+    // Signer A mints 20 tokens to Signer C
+    await expect(deployedERC20NoData.connect(deployerSignerA).mintNoData(signerC.address, 20))
+      .to.emit(deployedERC20NoData, 'Transfer')
+      .withArgs(ZERO_ADDRESS, signerC.address, 20);
+    // Signer B attempts to burn tokens from Signer A wallet (not allowed)
+    await expect(
+      deployedERC20NoData.connect(signerB).burnNoData(deployerSignerA.address, 10),
+    ).to.be.revertedWith('ERC20NoData: caller is not owner');
+    // Signer C attempts to burn tokens from Signer B wallet (not allowed)
```

Comment:
attempts to burn from signer B and C seem redundant, maybe changing one of them to signer A since A has "special power" as the contract owner but should still not be able to burn someone else's tokens